File: csharp_body_deref/Scripts/CustomLogic/CustomLogicBinopExpressionAst.cs
// ==================== AoTTG2 cross-reference ====================
// Type: CustomLogic.CustomLogicBinopExpressionAst
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/CustomLogic/CustomLogicBinopExpressionAst.c
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicBinopExpressionAst.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace CustomLogic;

[Token(Token = "0x200028A")]
internal class CustomLogicBinopExpressionAst : CustomLogicBaseExpressionAst
{
	[Token(Token = "0x4000DC5")]
	[FieldOffset(Offset = "0x18")]
	public CustomLogicBaseExpressionAst Left;

	[Token(Token = "0x4000DC6")]
	[FieldOffset(Offset = "0x20")]
	public CustomLogicBaseExpressionAst Right;

	[Token(Token = "0x4000DC7")]
	[FieldOffset(Offset = "0x28")]
	public CustomLogicToken Token;

	[Token(Token = "0x6000F71")]
	[Address(RVA = "0x3FB8430", Offset = "0x3FB8430", VA = "0x3FB8430")]
	public CustomLogicBinopExpressionAst(CustomLogicToken token, int line)
	{
	}
}

File: cpp/CustomLogic/CustomLogicBinopExpressionAst.c
// Type: CustomLogic.CustomLogicBinopExpressionAst
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/CustomLogic/CustomLogicBinopExpressionAst.cs
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicBinopExpressionAst.cs
// --------------------------------

// CustomLogic.CustomLogicBinopExpressionAst$$.ctor
// il2cpp: void CustomLogic_CustomLogicBinopExpressionAst___ctor (CustomLogic_CustomLogicBinopExpressionAst_o* __this, CustomLogic_CustomLogicToken_o* token, int32_t line, const MethodInfo* method);
// 0x40b8430

void CustomLogic_CustomLogicBinopExpressionAst___ctor
               (CustomLogic_CustomLogicBinopExpressionAst_o *__this,CustomLogic_CustomLogicToken_o *token,
               int32_t line,MethodInfo *method)

{
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields).Type = 9;
  (__this->fields).Line = line;
  (__this->fields).Token = token;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Token,token);
  return;
}


